apps/judicial-system/web/src/components/ProsecutorSelection/ProsecutorSelection.tsx

```diff
@@ -0,0 +1,78 @@
+import React, { useContext, useMemo } from 'react'
+import { useIntl } from 'react-intl'
+import { useQuery } from '@apollo/client'
+
+import { Select, Option } from '@island.is/island-ui/core'
+import {
+  isIndictmentCase,
+  User,
+  UserRole,
+} from '@island.is/judicial-system/types'
+import { FormContext } from '@island.is/judicial-system-web/src/components/FormProvider/FormProvider'
+import { strings } from './ProsecutorSelection.strings'
+import { ProsecutorSelectionUsersQuery } from './prosecutorSelectionUsersGql'
+import { OptionsType, ValueType } from 'react-select'
+import { ReactSelectOption } from '@island.is/judicial-system-web/src/types'
+
+interface Props {
+  onChange: (prosecutorId: string) => boolean
+}
+
+const ProsecutorSelection: React.FC<Props> = (props) => {
+  const { onChange } = props
+  const { formatMessage } = useIntl()
+  const { workingCase } = useContext(FormContext)
+
+  const selectedProsecutor = useMemo(() => {
+    return workingCase.prosecutor
+      ? {
+          label: workingCase.prosecutor.name,
+          value: workingCase.prosecutor.id,
+        }
+      : undefined
+  }, [workingCase.prosecutor])
+
+  const { data, loading } = useQuery(ProsecutorSelectionUsersQuery, {
+    fetchPolicy: 'no-cache',
+    errorPolicy: 'all',
+  })
+
+  const availableProsecutors: OptionsType<Option> = useMemo(() => {
+    return data?.users
+      .filter(
+        (aUser: User) =>
+          aUser.role === UserRole.PROSECUTOR &&
+          (!workingCase.creatingProsecutor ||
+            aUser.institution?.id ===
+              workingCase.creatingProsecutor?.institution?.id),
+      )
+      .map((prosecutor: User) => ({
+        label: prosecutor.name,
+        value: prosecutor.id,
+      }))
+  }, [data?.users, workingCase.creatingProsecutor])
+
+  return (
+    <Select
+      name="prosecutor"
+      label={formatMessage(strings.label, {
+        isIndictmentCase: isIndictmentCase(workingCase.type),
+      })}
+      placeholder={formatMessage(strings.placeholder, {
+        isIndictmentCase: isIndictmentCase(workingCase.type),
+      })}
+      value={selectedProsecutor}
+      options={availableProsecutors}
+      onChange={(value: ValueType<Option>) => {
+        const id = (value as ReactSelectOption).value
+        if (id && typeof id === 'string') {
```

Comment:
Why are we checking the type of the id here?

Reply:
Select component typing in island-ui is pretty bad. `value` is typed `number | string` even though it is always a number in this case. This check is only shut up typescript.